ambari_client.rb

```diff
@@ -181,6 +181,8 @@ def remove_component(cluster:, host:, component:)
   end
 
   def add_host(cluster:, host:)
+    uri = host(cluster: cluster, host: host)['href']
+    RestClient::Request.new(:method => :put, :url => uri, :user => @user, :password => @password).execute
```

Comment:
Have you tested this? I would assume this won't work but haven't tested it yet myself, because you are asking for the href of a nonexistent host in this call. The correct URI to add a host is to the hosts collection which is one level above the href to an individual host like you're asking for here.